lib/chef/knife.rb

```diff
@@ -88,6 +88,7 @@ def self.reset_subcommands!
     end
 
     def self.inherited(subclass)
+     super
```

Comment:
By calling `super` we are leveraging the options inheritance behavior that [mixlib-cli](https://github.com/chef/mixlib-cli/blob/2e77705cc598188ad80d6130df79ffb19781fb18/lib/mixlib/cli.rb#L44) has. @tas50 @lamont-granquist Do y'all know of any issues this will cause? If it will we can instead switch do doing the options merging that other classes were doing.

The removal of those from other files is because they are no longer necessary. This default behavior of inheriting options _seems_ more correct to me.

Reply:
@Vasu1105 CI is failing because of ChefStyle - this `super` needs 1 more space in front of it

Reply:
Thank you @tyler-ball. Fixed the chef style.